### DISCUSSION TITLE
React App with Fief

### DISCUSSION THREAD

**Discussion by bnewman-tech** [fief-dev/fief#111] [Q&A] [answered]
Howdy, 

I have been trying to set up a react app with the guide below. 
I have tried using CRA and Vite with typescript and vanilla. 
I have been unsuccessful with both. 

Any thoughts? Do you have a sample react project to try out?

https://docs.fief.dev/integrate/javascript/frontend/react/

For CRA: 
![image](https://user-images.githubusercontent.com/60909819/200971416-849b17f3-c7ce-4504-953b-271be3519ad5.png)

For Vite: 
![image](https://user-images.githubusercontent.com/60909819/200971551-31358954-65c4-4b27-a5e9-4b3a42862652.png)

**Comment by fief-bailiff** [comment]
Hail, @bnewman20 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
There's indeed a bundling problem with the library 🤔 I'm looking into this!

**Reply by su-network** [reply]
@frankie567 I got the same problem when using Fief in my CRA or in Vite But it got solved after installing vite plugin node polyfills In vite react app.
Link : https://socket.dev/npm/package/vite-plugin-node-polyfills

**Reply by bnewman-tech** [reply]
That polyfill plugin also worked for Vite. Thank you!!

**Comment by frankie567** [comment, marked as answer]
I _think_ I've managed to fix the issue with [`v0.10.2`](https://github.com/fief-dev/fief-js/releases/tag/v0.10.2) of the JavaScript SDK.

It should correctly start a CRA without complaining about `crypto` (no need to change anythin in your webpack config).

Tell me if it works for you!

**Reply by bnewman-tech** [reply]
Yes, that works! Thank you!!
